tests/gql_behave/tests/memgraph_V1/features/cartesian.feature

```diff
@@ -172,6 +172,111 @@ Feature: Cartesian
             """
         Then the result should be empty
 
+    Scenario: Multiple match with WHERE x = y 01
+        Given an empty graph
+        And having executed
+            """
+            CREATE (:A {id: 1}), (:A {id: 2}), (:B {id: 1})
+            """
+        When executing query:
+            """
+            MATCH (a:A) MATCH (b:B) WHERE a.id = b.id RETURN a, b
+            """
+        Then the result should be:
+            | a            | b            |
+            | (:A {id: 1}) | (:B {id: 1}) |
+
+    Scenario: Multiple match with WHERE x = y 02
+        Given an empty graph
+        And having executed
+            """
+            CREATE (:A {id: 1}), (:A {id: 2}), (:B {id: 1}), (:B {id: 2})
+            """
+        When executing query:
+            """
+            MATCH (a:A) MATCH (b:B) WHERE a.id = b.id RETURN a, b
+            """
+        Then the result should be:
+            | a            | b            |
+            | (:A {id: 1}) | (:B {id: 1}) |
+            | (:A {id: 2}) | (:B {id: 2}) |
+
+    Scenario: Multiple match with WHERE x = y 03
+        Given an empty graph
+        And having executed
+            """
+            CREATE (:A {prop: 1, id: 1}), (:A {prop: 2, id: 2}), (:A {prop: 1, id: 2}), (:B {prop: 2, id: 3})
+            """
+        When executing query:
+            """
+            MATCH (a) MATCH (b) WHERE a.prop = b.prop RETURN a, b
+            """
+        Then the result should be:
+            | a                     | b                     |
+            | (:A {id: 1, prop: 1}) | (:A {id: 1, prop: 1}) |
+            | (:A {id: 1, prop: 1}) | (:A {id: 2, prop: 1}) |
+            | (:A {id: 2, prop: 2}) | (:A {id: 2, prop: 2}) |
+            | (:A {id: 2, prop: 2}) | (:B {id: 3, prop: 2}) |
+            | (:A {id: 2, prop: 1}) | (:A {id: 1, prop: 1}) |
+            | (:A {id: 2, prop: 1}) | (:A {id: 2, prop: 1}) |
+            | (:B {id: 3, prop: 2}) | (:A {id: 2, prop: 2}) |
+            | (:B {id: 3, prop: 2}) | (:B {id: 3, prop: 2}) |
+
+
+    Scenario: Multiple match with WHERE x = y 04
+        Given an empty graph
+        And having executed
+            """
+            CREATE (:A {prop: 1, id: 1}), (:A {prop: 2, id: 2}), (:A {prop: 1, id: 2}), (:B {prop: 2, id: 3})
+            """
+        When executing query:
+            """
+            MATCH (a:A) MATCH (b:A) WHERE a.prop = b.prop RETURN a, b
+            """
+        Then the result should be:
+            | a                     | b                     |
+            | (:A {id: 1, prop: 1}) | (:A {id: 1, prop: 1}) |
+            | (:A {id: 1, prop: 1}) | (:A {id: 2, prop: 1}) |
+            | (:A {id: 2, prop: 2}) | (:A {id: 2, prop: 2}) |
+            | (:A {id: 2, prop: 1}) | (:A {id: 1, prop: 1}) |
+            | (:A {id: 2, prop: 1}) | (:A {id: 2, prop: 1}) |
```

Comment:
create a test with different props
e.g.
`MATCH (a:A) MATCH (b:A) WHERE a.prop = b.prop2 RETURN a, b`

Reply:
and reverse

Reply:
Added